### DISCUSSION TITLE
Performance Endpoints

### DISCUSSION THREAD

**Discussion by vmesel** [prest/prest#772] [API Server]
I really think we could add some endpoints for database hard users to lear how to manage better their databases. We could do something based on @turicas [presentation](https://www.youtube.com/watch?v=tS5f7Mo6lTA&ab_channel=pythonbrasil).

**Comment by vmesel** [comment]
I was thinking of adding an endpoint to add this query:

```
SELECT table_catalog as database,
       table_schema as schema,
       table_name,
       table_type as type,
       sub.total_size as total_size,
       sub.data_size as data_size,
       sub.indexes_size as indexes_size
       FROM information_schema.tables natural join (
    SELECT relname as table_name,
    pg_size_pretty(pg_total_relation_size(relid)) as total_size,
    pg_size_pretty(pg_relation_size(relid)) as data_size,
    pg_size_pretty(pg_indexes_size(relid)) as indexes_size
    FROM pg_catalog.pg_statio_user_tables
    ) sub
```

And others for alignment-checking and performance tuning.

**Comment by avelino** [comment]
@vmesel I like to have endpoints that give metrics about the connected bank, but I feel that this should not be part of prestd's core.

**Why not in the core?**
it is not essential endpoint and not all users would use it


> as a **[plugin](https://docs.prestd.com/prestd/docs/api-reference/plugins/)** it makes much more sense

**Reply by vmesel** [reply]
We could do it as the first official pREST plugin. What do you think?

**Reply by avelino** [reply]
I think it's great, we use it to think how to create the prestd marketplace (for official plugin distribution)
